pkg/operator/ceph/cluster/cluster.go

```diff
@@ -460,6 +466,17 @@ func (c *cluster) preMonStartupActions(cephVersion cephver.CephVersion) error {
 // It gets executed right after the main mon Start() method
 // Basically, it is executed between the monitors and the manager sequence
 func (c *cluster) postMonStartupActions() error {
+	// full cluster spec/status will be used to inform various cephx rotations
+	clusterObj := &cephv1.CephCluster{}
+	if err := c.context.Client.Get(c.ClusterInfo.Context, c.ClusterInfo.NamespacedName(), clusterObj); err != nil {
+		return errors.Wrapf(err, "failed to get cluster %v.", c.ClusterInfo.NamespacedName())
+	}
```

Comment:
@sp98 please integrate this into the mon key rotation work as noted here: https://github.com/rook/rook/pull/16176/files#r2267985572